Tasks/Maven/maventask.ts

```diff
@@ -147,6 +150,49 @@ async function execBuild() {
             var mvnRun = tl.tool(mvnExec);
             mvnRun.arg('-f');
             mvnRun.arg(mavenPOMFile);
+            mvnRun.arg('help:effective-pom');
+            return util.collectFeedRepositoriesFromEffectivePom(mvnRun.execSync()['stdout'])
+            .then(function (repositories) {
+                tl.debug('Repositories: ' + JSON.stringify(repositories));
+                settingsXmlFile = path.join(os.tmpdir(), 'settings.xml');
+
+                tl.debug('checking to see if there are settings.xml in use');
+                let options: RegExpMatchArray = mavenOptions ? mavenOptions.match(/([^" ]*("[^"]*")[^" ]*)|[^" ]+/g) : undefined;
+                if (options) {
+                    mavenOptions = '';
+                    for (let i = 0; i < options.length; ++i) {
+                        if ((options[i] === '--settings' || options[i] === '-s') && (i + 1) < options.length) {
+                            i++; // increment to the file name
+                            let suppliedSettingsXml: string = options[i];
+                            tl.cp(path.join(tl.cwd(), suppliedSettingsXml), settingsXmlFile, '');
+                            tl.debug('using settings file: ' + settingsXmlFile);
+                        } else {
+                            if (mavenOptions) {
+                                mavenOptions = mavenOptions.concat(' ');
+                            }
+                            mavenOptions = mavenOptions.concat(options[i]);
+                        }
+                    }
+                }
+                return util.mergeCredentialsIntoSettingsXml(settingsXmlFile, repositories);
+            })
+            .fail(function (err) {
+                return Q.reject(err);
+            });
+        })
+        .fail(function (err) {
+            console.error(err.message);
```

Comment:
I'm inclined to leave this one alone.  The rest of this module uses console.error (and I copied its use).  I'm thinking change them all or be consistent.